### PR TITLE
fix(dot/state): fix a bug in IsDescendantOf

### PR DESCRIPTION
**Pull request by kishansagathiya**
## Changes
for loop inside IsDescendantOf was missing logic to iterate to next data, this PR fixes that.

Raising this against `fix/kishan/key-not-found` for easier testing and better diffs. Will move this against `development` once `fix/kishan/key-not-found` is merged with `development`

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Primary Reviewer

@EclesioMeloJunior

### CODE REVIEW

File: dot/state/block.go [hunk 1]
```diff
@@ -773,10 +773,14 @@ func (bs *BlockState) IsDescendantOf(ancestor, descendant common.Hash) (bool, er
 			return false, fmt.Errorf("getting header: %w", err2)
 		}
 
-		for current := descendantHeader; descendantHeader.Number < ancestorHeader.Number; {
+		for current := descendantHeader; current.Number < ancestorHeader.Number; {
```
Comment (EclesioMeloJunior) [inline dot/state/block.go:776]:
```suggestion
		for current := descendantHeader; ancestorHeader.Number < current.Number; {
```
The descendant must not have a block number lower than the ancestor

File: dot/state/block.go [hunk 2]
```diff
@@ -773,10 +773,14 @@ func (bs *BlockState) IsDescendantOf(ancestor, descendant common.Hash) (bool, er
 			return false, fmt.Errorf("getting header: %w", err2)
 		}
```
Comment (EclesioMeloJunior) [inline dot/state/block.go:775]:
```suggestion
if descendantHeader.Number < ancestorHeader.Number {
  return errors.New("descendent lower than ancestor")
}
```

File: dot/state/block.go [hunk 3]
```diff
@@ -773,10 +773,14 @@ func (bs *BlockState) IsDescendantOf(ancestor, descendant common.Hash) (bool, er
...
 
-		for current := descendantHeader; descendantHeader.Number < ancestorHeader.Number; {
+		for current := descendantHeader; current.Number < ancestorHeader.Number; {
 			if current.ParentHash == ancestor {
 				return true, nil
 			}
+			current, err2 = bs.GetHeader(current.ParentHash)
```
Comment (EclesioMeloJunior) [inline dot/state/block.go:780]:
```suggestion
			current, err = bs.GetHeader(current.ParentHash)
```

Comment (kishansagathiya) [reply dot/state/block.go:780]: discussed over meeting, decided to ignore it

**Review by EclesioMeloJunior — APPROVED** [review]
lgtm!

**Review by jimjbrettj — APPROVED** [review]
lgtm

**Pull request merged**